1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,21 @@
-function setAlarm() {}
+function setAlarm() {
+  const btnSet = document.createElement("set");
+  const MyTimeremaining = document.getElementById("timeRemaining");
+  const MyInputAlarmSet = document.getElementById("alarmSet");
+  let myDate = new Date("2017-01-01 00:00:00");
+  myDate.setSeconds(MyInputAlarmSet.value);
+  MyTimeremaining.innerHTML =
+    "Time Remaining : " + myDate.toString().substring(19, 24);
+  let myInterval = setInterval(() => {
+    myDate.setSeconds(myDate.getSeconds() - 1);
+    MyTimeremaining.innerHTML =
+      "Time Remaining" + myDate.toString().substring(19, 24);
+    if (myDate.toString().substring(19, 24) === "00:00") {
+      playAlarm();
```

Comment:
An easier way of getting the seconds would be `myDate.getSeconds()`
So you dont have to extract it from the string

Reply:
Good idea using a date though!

Reply:
thanks